prow/scripts/cluster-integration/kyma-integration-gardener-eventing-upgrade.sh

```diff
@@ -0,0 +1,169 @@
+export TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS="${TEST_INFRA_SOURCES_DIR}/prow/scripts/cluster-integration/helpers"
+export EVENTMESH_SECRET_FILE="${CREDENTIALS_DIR}/serviceKey" # For eventing E2E fast-integration tests
+
+# shellcheck source=prow/scripts/lib/log.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/log.sh"
+# shellcheck source=prow/scripts/lib/utils.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/utils.sh"
+# shellcheck source=prow/scripts/lib/kyma.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/kyma.sh"
+# shellcheck source=prow/scripts/cluster-integration/helpers/eventing.sh
+source "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/eventing.sh"
+# shellcheck source=prow/scripts/lib/gardener/gardener.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/gardener.sh"
+
+# All provides require these values, each of them may check for additional variables
+requiredVars=(
+    BOT_GITHUB_TOKEN
+    GARDENER_PROVIDER
+    KYMA_PROJECT_DIR
+    GARDENER_REGION
+    GARDENER_ZONES
+    GARDENER_CLUSTER_VERSION
+    GARDENER_KYMA_PROW_KUBECONFIG
+    GARDENER_KYMA_PROW_PROJECT_NAME
+    GARDENER_KYMA_PROW_PROVIDER_SECRET_NAME
+    CREDENTIALS_DIR
+    EVENTMESH_SECRET_FILE
+)
+
+utils::check_required_vars "${requiredVars[@]}"
+log::info "### Starting pipeline"
+
+if [[ $GARDENER_PROVIDER == "azure" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/azure.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/azure.sh"
+elif [[ $GARDENER_PROVIDER == "aws" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/aws.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/aws.sh"
+elif [[ $GARDENER_PROVIDER == "gcp" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/gcp.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/gcp.sh"
+else
+    log::error "GARDENER_PROVIDER ${GARDENER_PROVIDER} is not yet supported"
+    exit 1
+fi
+
+# nice cleanup on exit, be it successful or on fail
+trap gardener::cleanup EXIT INT
+
+#Used to detect errors for logging purposes
+ERROR_LOGGING_GUARD="true"
+export ERROR_LOGGING_GUARD
+
+RANDOM_NAME_SUFFIX=$(LC_ALL=C tr -dc 'a-z0-9' < /dev/urandom | head -c6)
+readonly COMMON_NAME_PREFIX="grd"
+COMMON_NAME=$(echo "${COMMON_NAME_PREFIX}${RANDOM_NAME_SUFFIX}" | tr "[:upper:]" "[:lower:]")
+export COMMON_NAME
+
+### Cluster name must be less than 10 characters!
+export CLUSTER_NAME="${COMMON_NAME}"
+
+# set pipefail to handle right errors from tests
+set -o pipefail
+
+# Install Kyma form latest release
+kyma::get_last_release_version -t "${BOT_GITHUB_TOKEN}"
+
+export KYMA_SOURCE="${kyma_get_last_release_version_return_version:?}"
+log::info "### Reading release version from RELEASE_VERSION file, got: ${KYMA_SOURCE}"
+
+# checks required vars and initializes gcloud/docker if necessary
+gardener::init
+
+# if MACHINE_TYPE is not set then use default one
+gardener::set_machine_type
+
+kyma::install_cli_last_release
```

Comment:
I don't think the Kyma CLI version that you get here, actually has the latest reconciler. Maybe this is the reason your cluster is unhealthy after upgrade.

Reply:
I replaced it with latest stable cli version, there is also unstable version, but I wouldn't rely on that, because in case the unstable version is broken, it will break out whole prow pipeline

Reply:
I think we should consider using the unstable version, because the stable tag on Kyma CLI  tends to be behind main head (see: https://github.com/kyma-project/cli/tree/stable) which means we are testing using an old commit of the CLI/Reconciler which is in my opinion not effective.

Reply:
follow the https://github.com/kyma-project/test-infra/pull/4581#discussion_r786136228